mozbench/mozbench.py

```diff
@@ -267,25 +277,49 @@ def cli(args):
         use_android = False
 
     if use_android:
+        logger.info('prepare for installing fennec')
         fennec_pkg_name = get_fennec_pkg_name(args.firefox_path)
-        firefox_binary = install_fennec(logger, args.firefox_path,
-                                        fennec_pkg_name, args.device_serial)
+        success = install_fennec(logger, args.firefox_path, fennec_pkg_name,
+                                 args.device_serial)
+        if not success:
+            logger.error('fennec installation fail')
+            return 1
+        logger.info('fennec installation succeed')
     else:
         if args.run_android_browser:
-            logger.warning('Stock Android browser only supported on Android')
+            logger.warning('stock Android browser only supported on Android')
         if args.run_dolphin:
-            logger.warning('Dolphin browser only supported on Android')
+            logger.warning('dolphin browser only supported on Android')
 
-    logger.info('starting webserver on %s' % args.test_host)
     static_path = os.path.abspath(os.path.join(os.path.dirname(__file__),
                                                'static'))
-    httpd = wptserve.server.WebTestHttpd(host=args.test_host, port=8888,
-                                         routes=routes, doc_root=static_path)
-    httpd.start()
+    # start http server and request handler
+    httpd = None
+    while httpd is None:
+        port = 10000 + random.randrange(0, 50000)
+        if args.test_port:
+            port = int(args.test_port)
```

Comment:
If it turns out test_port is in use this will loop forever. I have had problems in the past with processes hanging on to the test_port after they should have released it. I think we would be better off with:
if args.test_port:
    ...
else:
    while httpd is None:
         ...